drivers/gpio/Kconfig.gd32

```diff
@@ -5,5 +5,6 @@ config GPIO_GD32
 	bool "GD32 GPIO driver"
 	default y
 	depends on DT_HAS_GD_GD32_GPIO_ENABLED
+	select GD32_EXTI
```

Comment:
EXTI provide some basic capability for Zephyr, thread schedule, error handle, external event receive, etc. GPIO is an optional one.
There no tightly relation between them. So I think we can ignore the depends for these, EXTI will always accessable.